src/main/java/jenkins/plugins/publish_over_ssh/BapSshClient.java

```diff
@@ -233,6 +234,148 @@ private void deleteFile(final String pathName) {
         }
     }
 
+    private String getStringFromArrayList(ArrayList<String> commandArguments) {
+        StringBuilder command = new StringBuilder();
+        for (String str : commandArguments) {
+            command.append(str);
+            command.append(" ");
+        }
+        return command.toString().trim();
+    }
```

Comment:
Why not just use `string.join` for this? https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-

Reply:
Is it right to use methods from Java 8, when java level is 7 in pom.xml?

Reply:
Ok, please update to use the Java 8 methods to simplify these items.